docs/releases/slashing-release/interfaces/core/slashing-parameter-configuration.md

```diff
@@ -63,12 +53,16 @@ interface IAVSDirectory {
     /// EXTERNAL - STATE MODIFYING
     
     /**
-     * @notice Modifies the proportions of slashable stake allocated to a list of operatorSets for a set of strategies
+     * @notice Modifies the propotions of slashable stake allocated to a list of operatorSets for a set of strategies
+     * This will revert if the operator has never set their allocationDelay before.
      * @param operator address to modify allocations for
      * @param allocations array of magnitude adjustments for multiple strategies and corresponding operator sets
      * @param operatorSignature signature of the operator if msg.sender is not the operator
-     * @dev updates freeMagnitude for the updated strategies
-     * @dev must be called by the operator
+     * @dev Updates freeMagnitude for the updated strategies
+     * @dev Must be called by the operator or with a valid operator signature
+     * @dev For each allocation, allocation.operatorSets MUST be ordered in ascending order according to the
+     * encoding of the operatorSet. This is to prevent duplicate operatorSets being passed in. The easiest way to ensure
```

Comment:
ew, this is kind of gross to handle offchain, @shrimalmadhur what would you prefer

Reply:
we also have to think about LRT or every other client using this directly. I think this should be baked in solidity if possible - otherwise every other thing built on top has to think about it.